helper_functions.py

```diff
@@ -1,52 +1,124 @@
 from flask import (Flask, request, render_template, flash, session, jsonify, abort)
 from model import *
 from sqlalchemy import desc
+#for facebook environmental variable
+import os
+import facebook
 
 
-##### '/' helper functions #####
-def gather_all_notes_from_db():
+##### session helper functions #####
+def current_user():
+    """ Return the user object if in session """
 
-    return Note.query.order_by(desc(Note.created_at)).all()
+    if 'current_user' in session:
 
-##### '/add_note' helper functions #####
-def commit_note_to_db(note_title, new_note):
+        return User.query.get(session['current_user'])
+
+    else:
+
+        return None
+
+
+###### facebook sign in helper functions #####
+def facebook_app_id():
+    """ Retrieve app_id from local environment """
+
+    app_id=os.environ["APP_ID"]
+
+    return app_id
+
+####### GET '/notes' helper functions ########
+def gather_all_notes_from_db(user_id):
+    """Gather all notes by user id """
+    
+    note = Note.query.filter_by(user_id=user_id).order_by(desc(Note.created_at)).all()
+
+    if not note:
+
+        return None
+
+    return note
```

Comment:
As is, this can/should be a one-liner.

Consider having a second argument for the ordering and default it to 'desc'.

Ordering by `id` is the same as ordering by `created_at` only way, way faster.